doc/allocation.md

```diff
@@ -106,8 +106,167 @@ The normal `improve()` step selects the next higher layer if there is sufficient
 The preferred resolution [can be configured](https://github.com/jitsi/jitsi-videobridge/blob/master/jvb/src/main/resources/reference.conf#L40).
 
 # Signaling
-TODO
-## Old message format
-TODO
+This section describes the signaling between the client and the bridge that affects bandwidth allocation.
+
+## Legacy message format
+This is the signaling currently used in jitsi-meet, but the intention is to replace it with the new format and
+eventually deprecate this format.
+
+This format is not expressive enough to enable all features supported by the bridge, so some assumptions have to be
+made. Notably:
+1. The selection strategy is inferred from the number of selected endpoints (`TileView` if more than one endpoints
+is selected, and `StageView` otherwise).
+2. Multiple selected endpoints are not supported with the usual semantics. When multiple endpoints are selected, this
+signals the use of `TileView`, but does NOT override the speaker order.
+3. Constraints are constructed solely based on `maxFrameHeight` and the selected endpoints.
+### LastN
+LastN is set with a `LastNChangedEvent` message:
+```json
+{
+  "colibriClass": "LastNChangedEvent",
+  "lastN": 3
+}
+```
+
+### MaxFrameHeight
+A global maximum resolution is set with a `ReceiverVideoConstraint` message:
+```json
+{
+  "colibriClass": "ReceiverVideoConstraint",
+  "maxFrameHeight": 360
+}
+```
+
+### Selected endpoints
+Selected endpoints are set with a `SelectedEndpointsChangedEvent` message:
+```json
+{
+  "colibriClass": "SelectedEndpointsChangedEvent",
+  "selectedEndpoints": ["A", "B"]
+}
+```
+
 ## New message format
-TODO
+The new format uses a single message with a set fields:
```

Comment:
Grammar, "with set fields" or "with a set of fields" depending on what you meant